ba/cheatsheets/index.md

```diff
@@ -0,0 +1,86 @@
+|  `(1 to 5).map { x => val y=x*2; println(y); y }`                                                             |  anonimna funkcija: blokovski stil vraća vrijednost zadnjeg izraza. |
+|  `(1 to 5) filter {_%2 == 0} map {_*2}`                                                                  |  anonimne funkcije: pipeline stil. (može i sa oblim zagradama). |
+|  `def compose(g:R=>R, h:R=>R) = (x:R) => g(h(x))` <br> `val f = compose({_*2}, {_-1})`                   |  anonimne funkcije: da bi proslijedili više blokova, potrebne su dodatne zagrade. |
+|  `val zscore = (mean:R, sd:R) => (x:R) => (x-mean)/sd`                                                   |  curry-jevanje, očita sintaksa. |
+|  `def zscore(mean:R, sd:R) = (x:R) => (x-mean)/sd`                                                       |  curry-jevanje, očita sintaksa. |
+|  `def zscore(mean:R, sd:R)(x:R) = (x-mean)/sd`                                                           |  curry-jevanje, sintaksni šećer(kratica). ali onda: |
+|  `val normer = zscore(7, 0.4) _`                                                                          |  je potrebna prateća donja crta za parcijalnu primjenu, samo kod šećer(skraćene) verzije. |
+|  `def mapmake[T](g:T=>T)(seq: List[T]) = seq.map(g)`                                                     |  generički tip. |
+|  `5.+(3); 5 + 3` <br> `(1 to 5) map (_*2)`                                                               |  infiksni šećer. |
+|  `def sum(args: Int*) = args.reduceLeft(_+_)`                                                            |  varirajući broj argumenata(varargs). |
+|  <span id="packages" class="h2">paketi</span>                                                                         |                 |
+|  `import scala.collection._`                                                                             |  džoker(wildcard) import. |
+|  `import scala.collection.Vector` <br> `import scala.collection.{Vector, Sequence}`                      |  selektivni import. |
+|  `import scala.collection.{Vector => Vec28}`                                                             |  preimenujući import. |
+|  `import java.util.{Date => _, _}`                                                                       |  import svega iz java.util paketa osim Date. |
+|  `package pkg` _na početku fajla_ <br> `package pkg { ... }`                                             |  deklaracija paketa. |
+|  <span id="data_structures" class="h2">strukture podataka</span>                                                           |                 |
+|  `(1,2,3)`                                                                                               |  n-torka(tuple) literal. (`Tuple3`) |
+|  `var (x,y,z) = (1,2,3)`                                                                                 |  destrukturirajuće vezivanje: otpakivanje n-torke podudaranjem uzorka(pattern matching). |
```

Comment:
I think that "destruktivno" would refer to "destroying" and not to "destructuring". We are not destroying anything with pattern matching so I left that as before.